### ISSUE TITLE
TypeError

### ISSUE THREAD

**Issue by joelgombin** [goldingn/greta#45]
I tried to replicate the example in the README, but failed, even though TensorFlow is correctly installed (I can succesfully replicate the examples from the TensorFlow documentation). 

```r
library(greta)
#> 
#> Attachement du package : 'greta'
#> The following objects are masked from 'package:stats':
#> 
#>     binomial, poisson
#> The following objects are masked from 'package:base':
#> 
#>     %*%, beta, diag, gamma, sweep

# create parameters, stating their prior distributions
intercept = normal(0, 5)
coefficient = normal(0, 3)
sd = lognormal(0, 3)

# write the equation for the expected mean sepal length
mean <- intercept + coefficient * iris$Petal.Length

# define the likelihood of the observed data
likelihood(iris$Sepal.Length) = normal(mean, sd)

model <- define_model(intercept, coefficient, sd)
#> Error in py_call_impl(callable, dots$args, dots$keywords): TypeError: Expected int32, got list containing Tensors of type '_Message' instead.
#> 
#> Detailed traceback: 
#>   File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/ops/array_ops.py", line 1000, in concat
#>     dtype=dtypes.int32).get_shape(
#>   File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/ops.py", line 669, in convert_to_tensor
#>     ret = conversion_func(value, dtype=dtype, name=name, as_ref=as_ref)
#>   File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/constant_op.py", line 176, in _constant_tensor_conversion_function
#>     return constant(v, dtype=dtype, name=name)
#>   File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/constant_op.py", line 165, in constant
#>     tensor_util.make_tensor_proto(value, dtype=dtype, shape=shape, verify_shape=verify_shape))
#>   File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/tensor_util.py", line 367, in make_tensor_proto
#>     _AssertCompatible(values, dtype)
#>   File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/tensor_util.py", line 302, in _AssertCompatible
#>     (dtype.name, repr(mismatch), type(mismatch).__name__))
```

Here is my session info:

```r
devtools::session_info()
#> Session info --------------------------------------------------------------
#>  setting  value                       
#>  version  R version 3.4.0 (2017-04-21)
#>  system   x86_64, linux-gnu           
#>  ui       X11                         
#>  language fr_FR                       
#>  collate  fr_FR.UTF-8                 
#>  tz       Europe/Paris                
#>  date     2017-04-30
#> Packages ------------------------------------------------------------------
#>  package    * version    date       source                            
#>  backports    1.0.5      2017-01-18 CRAN (R 3.3.2)                    
#>  coda         0.19-1     2016-12-08 cran (@0.19-1)                    
#>  devtools     1.12.0     2016-12-05 CRAN (R 3.4.0)                    
#>  digest       0.6.12     2017-01-27 CRAN (R 3.4.0)                    
#>  evaluate     0.10       2016-10-11 CRAN (R 3.3.1)                    
#>  greta      * 0.1.1      2017-04-30 Github (goldingn/greta@bd64983)   
#>  htmltools    0.3.6      2017-04-28 CRAN (R 3.4.0)                    
#>  knitr        1.15.1     2016-11-22 CRAN (R 3.3.2)                    
#>  lattice      0.20-35    2017-03-25 CRAN (R 3.3.3)                    
#>  magrittr     1.5        2014-11-22 CRAN (R 3.4.0)                    
#>  memoise      1.1.0      2017-04-21 CRAN (R 3.4.0)                    
#>  R6           2.2.0      2016-10-05 CRAN (R 3.4.0)                    
#>  Rcpp         0.12.10    2017-03-19 CRAN (R 3.4.0)                    
#>  reticulate   0.7        2017-03-14 CRAN (R 3.3.3)                    
#>  rmarkdown    1.4.0.9001 2017-04-10 Github (rstudio/rmarkdown@b7434dc)
#>  rprojroot    1.2        2017-01-16 CRAN (R 3.3.2)                    
#>  stringi      1.1.5      2017-04-07 cran (@1.1.5)                     
#>  stringr      1.2.0      2017-02-18 cran (@1.2.0)                     
#>  tensorflow   0.7        2017-03-22 CRAN (R 3.3.3)                    
#>  withr        1.0.2      2016-06-20 CRAN (R 3.4.0)                    
#>  yaml         2.1.14     2016-11-12 CRAN (R 3.3.2)
```

**Comment by goldingn** [comment]
Huh, that's strange. Thanks for letting me know!

Would you mind checking what version of python and of the TensorFlow python module you have installed?

You should be able to do something like this in a terminal:
```
python --version
```
```
python -c 'import tensorflow as tf; print(tf.__version__)'
```

**Comment by goldingn** [comment]
note to myself: 
I tried and failed to replicate this error on my mac by rolling tensorflow & reticulate back to 0.7, R6 to 2.2.0, and trying tensorflow 1.1.0, 1.0.1 and 1.0.0

With 0.9.0 I get this different error: 
```
Error in py_get_attr_impl(x, name, silent) : 
  AttributeError: 'module' object has no attribute 'subtract'
```
which is due to a major API change at v1.0.0

So it's looking like something OS-dependent, or some other weird interaction.

Googling the error, I see quite a few discussions where no one can replicate this, and the winning suggestion is generally 'update tensorflow'.

This one suggests it's change to the order of arguments for `tf.concat` in 1.0.0:
http://stackoverflow.com/questions/41813665/tensorflow-slim-typeerror-expected-int32-got-list-containing-tensors-of-type

**Comment by rdrr1990** [comment]
I hit the same issue as @joelgombin on my Mac using Python 3.5 and tensorflow 0.12.0 but upgrading to tf 1.1.0 solved the problem--the README example runs just fine now. In case it helps, I am using R 3.3.3 "Another Canoe" and `library(greta)` works just fine now in RStudio 1.0.136 too.

**Comment by goldingn** [comment]
Ah, that's really helpful thanks!

So upgrading to >1.0.0 should fix this across the board. I'll amend the readme to be explicit about that.

**Comment by joelgombin** [comment]
OK, I updated tf and now everything works fine. Thanks!

**Comment by goldingn** [comment]
Great!

fixed [here](7bda762283026aed116d9804a8f0b178ec25e4cc) (for future reference)
